neqo-common/src/datagram.rs

```diff
@@ -7,20 +7,129 @@
 use std::net::SocketAddr;
 use std::ops::Deref;
 
+use enum_map::Enum;
+
 use crate::hex_with_len;
 
-#[derive(PartialEq, Eq, Clone)]
+// ECN (Explicit Congestion Notification) codepoints mapped to the
+// lower 2 bits of the TOS field.
+// https://www.iana.org/assignments/dscp-registry/dscp-registry.xhtml
+#[derive(Copy, Clone, PartialEq, Eq, Enum, Default)]
+#[repr(u8)]
+pub enum IpTosEcn {
+    #[default]
```

Comment:
I like enum-map, but it's a bit of a shame to have to pick it up for just this.  That should be a language-native feature.